### PR TITLE
Add reroll target button for 20% of the average skill reward

### PR DESCRIPTION
**Pull request by WAYN-Games**
### All Submissions

* [X] Can you post a screenshot of your changes (if applicable)?

image

### New Feature Submissions

* [X] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?
https://github.com/CryptoBlades/cryptoblades/issues/597

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

Add a button under "Choose New Weapon" to "Reroll Targets".
When clicking on "Reroll Targets", the player pay 20% (configurable through a setter) of the average reward he would have gotten from the fight for the maximum stamina available and get a new set of targets.
The change only affect the character/weapon couple.
The skill balance displayed on hte top rigth corner is not updated automatically (could not find how) but refreshing hte page show the skill is consumed.

The implementation meets the requirement but may not be optimal has I have limited experience in Solidity and Vue.js.

### CODE REVIEW

File: contracts/cryptoblades.sol [hunk 1]
```diff
@@ -68,6 +68,9 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
 
         // migrateTo_5e833b0
         durabilityCostFight = 1;
+		
+        // migrateTo_???????
+        rerollFightCost = 20;
```
Comment (raymond-h) [inline contracts/cryptoblades.sol:73]: This needs to go in a new `migrateTo_[X]`-style function. Name said new function `migrateTo_PLACEHOLDER` for now.

**Review by raymond-h — APPROVED** [review]

File: contracts/cryptoblades.sol (hunk 1 above)
Comment (raymond-h) [inline contracts/cryptoblades.sol:73]: This needs to go in a new `migrateTo_[X]`-style function. Name said new function `migrateTo_PLACEHOLDER` for now.

**Review by raymond-h — CHANGES_REQUESTED** [review]

**Comment by seiyria** [comment]
@raymond-h this one may need to wait. @bilbolPrime has a skill shop smoke bomb idea for this, although I'm not sure if it's been done yet.

**Comment by WAYN-Games** [comment]
@raymond-h , I think @seiyria refers to https://github.com/CryptoBlades/cryptoblades/pull/671
If I'm not mistaken, the smoke bomb from this PR will change the target for the character whatever the weapon.
Mine changes the target set for the couple character/weapon.

**Comment by raymond-h** [comment]
It sounds like there needs to be a discussion about this "smoke bomb" ordeal.

**Comment by seiyria** [comment]
Sorry, in the effort of reducing confusion we're going to go with the smoke bomb idea and work with that idea, as we want to rework most interactions that cost SKILL to use the SKILL shop somehow.

**Pull request closed without merging**